File: backend/src/github/github.controller.ts
import { Controller, Get, Param, Query } from '@nestjs/common';
import { GithubService } from './github.service';

@Controller('github')
export class GithubController {
  constructor(private readonly githubService: GithubService) {}

  @Get(':username/profile')
  getProfile(@Param('username') username: string) {
    return this.githubService.getProfile(username);
  }

  @Get(':username/repos')
  getRepos(@Param('username') username: string, @Query('limit') limit?: string) {
    return this.githubService.getRepos(username, limit ? parseInt(limit, 10) : undefined);
  }

  @Get(':username/summary')
  getSummary(@Param('username') username: string) {
    return this.githubService.getSummary(username);
  }
}

File: backend/src/main/java/com/devagent/github/GithubController.java
package com.devagent.github;

import org.springframework.web.bind.annotation.*;

import java.util.List;
import java.util.Map;

@RestController
@RequestMapping("/github")
public class GithubController {

    private final GithubService githubService;

    public GithubController(GithubService githubService) {
        this.githubService = githubService;
    }

    @GetMapping("/{username}/profile")
    public Map<String, Object> getProfile(@PathVariable String username) {
        return githubService.getProfile(username);
    }

    @GetMapping("/{username}/repos")
    public List<Map<String, Object>> getRepos(@PathVariable String username,
                                               @RequestParam(defaultValue = "100") int limit) {
        return githubService.getRepos(username, limit);
    }

    @GetMapping("/{username}/summary")
    public Map<String, Object> getSummary(@PathVariable String username) {
        return githubService.getSummary(username);
    }
}
